<!--
  Copyright 2018 The Cartographer Authors

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
-->

<launch>
  
      
  <!-- Cartographer -->
    <node pkg="cartographer_ros" type="cartographer_offline_node" name="cartographer_offline_node" ns="cartographer" args="
            -configuration_directory $(find cartographer_example)/config
            -configuration_basename config_3d.lua
            -urdf_filenames $(arg urdf_filenames)
            -bag_filenames $(arg bag_filenames)"
        output="screen">
        <remap from="scan" to="/scan"/>
        <remap from="echoes" to="/horizontal_laser_3d"/>
        <remap from="imu" to="/atlans_imu"/>
        <remap from="points2" to="/velodyne_points" />
        <remap from="points2_1" to="/horizontal_laser_3d" />
        <remap from="points2_2" to="/vertical_laser_3d" />
        <remap from="odom" to="/atlans_odom"/>
    </node>
      
  <node pkg="cartographer_ros" type="cartographer_occupancy_grid_node" name="cartographer_occupancy_grid_node" ns="cartographer" args="-resolution 0.05"/>
  
</launch>
